kdl_parser/CMakeLists.txt

```diff
@@ -1,51 +1,69 @@
 
 project(kdl_parser)
 
-find_package(Boost REQUIRED)
-include_directories(${Boost_INCLUDE_DIR})
-
-find_package(catkin REQUIRED
-  COMPONENTS roscpp rosconsole urdf cmake_modules
-)
+find_package(ament_cmake REQUIRED)
 find_package(orocos_kdl REQUIRED)
+find_package(tinyxml_vendor REQUIRED)
 find_package(TinyXML REQUIRED)
+find_package(urdf REQUIRED)
+find_package(urdfdom_headers REQUIRED)
+find_library(KDL_LIBRARY REQUIRED
+  NAMES orocos-kdl
+  HINTS ${orocos_kdl_LIBRARY_DIRS})
 
-include_directories(include ${orocos_kdl_INCLUDE_DIRS} ${TinyXML_INCLUDE_DIRS} ${catkin_INCLUDE_DIRS})
+include_directories(
+  include
+  ${orocos_kdl_INCLUDE_DIRS}
+  ${urdf_INCLUDE_DIRS}
+  ${urdfdom_headers_INCLUDE_DIRS}
+  ${TinyXML_INCLUDE_DIRS})
 
-link_directories(${catkin_LIBRARY_DIRS})
-link_directories(${Boost_LIBRARY_DIRS})
+link_directories(
+  ${orocos_kdl_LIBRARY_DIRS}
+  ${urdf_LIBRARY_DIRS})
 
-add_compile_options(-std=c++11)
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
 
-catkin_package(
-  LIBRARIES ${PROJECT_NAME} ${orocos_kdl_LIBRARIES}
-  INCLUDE_DIRS include
-  CATKIN_DEPENDS roscpp rosconsole urdf
-  DEPENDS orocos_kdl TinyXML
-)
+add_library(
+  ${PROJECT_NAME}
+  SHARED
+  src/kdl_parser.cpp)
+target_link_libraries(
+  ${PROJECT_NAME}
+  ${TinyXML_LIBRARIES}
+  ${orocos_kdl_LIBRARIES}
+  ${urdf_LIBRARIES})
 
-add_library(${PROJECT_NAME} src/kdl_parser.cpp)
-target_link_libraries(${PROJECT_NAME}
-  ${TinyXML_LIBRARIES} ${orocos_kdl_LIBRARIES} ${catkin_LIBRARIES}
-)
+if(WIN32)
+  target_compile_definitions(${PROJECT_NAME} PRIVATE "KDL_PARSER_BUILDING_DLL")
+endif()
 
-add_executable(check_kdl_parser src/check_kdl_parser.cpp )
-target_link_libraries(check_kdl_parser ${PROJECT_NAME})
+install(
+  TARGETS ${PROJECT_NAME}
+  LIBRARY DESTINATION lib
+  ARCHIVE DESTINATION lib
+  RUNTIME DESTINATION bin)
 
-if(CATKIN_ENABLE_TESTING)
-  find_package(catkin REQUIRED COMPONENTS rostest)
-  add_rostest_gtest(test_kdl_parser test/test_kdl_parser.launch test/test_kdl_parser.cpp )
-  target_link_libraries(test_kdl_parser ${PROJECT_NAME})
+install(
+  DIRECTORY include/${PROJECT_NAME}/
+  DESTINATION include/${PROJECT_NAME})
 
-  add_rostest_gtest(test_inertia_rpy test/test_inertia_rpy.launch test/test_inertia_rpy.cpp )
-  target_link_libraries(test_inertia_rpy ${PROJECT_NAME})
+if(BUILD_TESTING)
+  find_package(ament_cmake_cppcheck REQUIRED)
+  find_package(ament_cmake_cpplint REQUIRED)
+  find_package(ament_cmake_lint_cmake REQUIRED)
+  find_package(ament_cmake_uncrustify REQUIRED)
+  ament_cppcheck()
+  ament_cpplint()
+  ament_lint_cmake()
+  ament_uncrustify()
 endif()
 
-# How does CATKIN do this?
-#rosbuild_add_rostest(${PROJECT_SOURCE_DIR}/test/test_kdl_parser.launch)
-install(TARGETS ${PROJECT_NAME} 
-  DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION})
-
-install(DIRECTORY include/${PROJECT_NAME}/
-  DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION})
+ament_export_dependencies(orocos_kdl)
```

Comment:
this should export urdf as well as it's using urdf types in it's public API right?

Reply:
these dependencies should also be exported in the package.xml (or at least urdfdom_headers in the case of urdf)

Reply:
From looking at it, I think `orocos_kdl` and `urdfdom_headers` are the exported dependencies, while `urdf` is a build dependency.  I've marked them all as such in CMakeLists.txt and package.xml in 4c440e8

Reply:
>  I think orocos_kdl and urdfdom_headers are the exported dependencies

Yes that matches what I guessed when reviewing :+1: